public/pages/Dashboard/Components/AnomaliesLiveChart.tsx

```diff
@@ -187,12 +187,13 @@ export const AnomaliesLiveChart = (props: AnomaliesLiveChartProps) => {
         continue;
       }
       result.push({
-        [AD_DOC_FIELDS.DETECTOR_NAME]: !isEmpty(liveAnomalyData) ? '' : null,
+        [AD_DOC_FIELDS.DETECTOR_NAME]: !isEmpty(liveAnomalyData)
+          ? ''
+          : SPACE_STR,
```

Comment:
This is the only change for fix on the live chart issue

Reply:
Looks weird to set detector name as '' or ' ' , can you explain why we need an empty detector name or space as detector name ?

Reply:
When there exists liveAnomalyData, using '' the empty string makes sure those placeholder data won't be shown in legend because its name is ''; When liveAnomalyData does not exist, using space string as detector name can make sure data is plotted in the chart, and because in the case of no liveAnomalyData, legend is hidden, then we are able to see the live chart shown, even without actual anomaly data. Actually the SPACE_STR can be anything, since legend is hidden in that case.